tests/explainers/test_gradient.py

```diff
@@ -271,3 +271,62 @@ def forward(self, x1, x2):
     sums = np.array([shap_x1[i].sum() + shap_x2[i].sum() for i in range(len(shap_x1))])
     d = np.abs(sums - diff).sum()
     assert d / (np.abs(diff).sum()+0.01) < 0.1, "Sum of SHAP values does not match difference! %f" % (d / np.abs(diff).sum())
+
+def test_tf_pd_input(random_seed):
+    """ Test tabular (batch_size, features) pd.DataFrame input. """
+    pd = pytest.importorskip('pandas')
```

Comment:
Pandas is a non-optional dependency of the package, so it can be safely imported in the normal way at the top of the file